recommend/runtimePolicy.go

```diff
@@ -0,0 +1,150 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2022 Authors of KubeArmor
+
+package recommend
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+
+	opb "github.com/accuknox/auto-policy-discovery/src/protobuf/v1/observability"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+var ms MatchSpec
+var filePaths, processPaths SysRule
+
+// CreateRuntimePolicy function generates runtime policy for service account
+func CreateRuntimePolicy(o *Options, img *ImageInfo) error {
+	ms = MatchSpec{}
+	var label string
+
+	for _, labels := range o.Labels {
+		kvPair := strings.FieldsFunc(labels, labelSplitter)
+		label = strings.TrimPrefix(fmt.Sprintf("%s,%s=%s", label, kvPair[0], kvPair[1]), ",")
+
+	}
+
+	ms, err := summarySearch(label, o.Namespace)
+
+	img.writePolicyFile(ms)
+
+	return err
+
+}
+
+func summarySearch(labels, namespace string) (MatchSpec, error) {
+
+	gRPC := ""
+
+	if val, ok := os.LookupEnv("DISCOVERY_SERVICE"); ok {
+		gRPC = val
+	} else {
+		gRPC = "localhost:9089"
+	}
+
+	// create a client
+	conn, err := grpc.Dial(gRPC, grpc.WithTransportCredentials(insecure.NewCredentials()))
+	if err != nil {
+		return ms, errors.New("could not connect to the server. Possible troubleshooting:\n- Check if discovery engine is running\n- Create a portforward to discovery engine service using\n\t\033[1mkubectl port-forward -n explorer service/knoxautopolicy --address 0.0.0.0 --address :: 9089:9089\033[0m\n[0m")
+	}
+	defer conn.Close()
+
+	client := opb.NewObservabilityClient(conn)
+
+	podResp, err := client.GetPodNames(context.Background(), &opb.Request{
+		Label:     labels,
+		NameSpace: namespace,
+	})
+
+	if err != nil {
+		return ms, err
+	}
+
+	for _, podName := range podResp.PodName {
+
+		sumResp, err := client.Summary(context.Background(), &opb.Request{
+			PodName: podName,
+			Type:    "process",
+		})
+		if err != nil {
+			return ms, err
+		}
+		_ = checkProcessFileData(sumResp)
+		sumResp, err = client.Summary(context.Background(), &opb.Request{
+			PodName: podName,
+			Type:    "file",
+		})
+		if err != nil {
+			return ms, err
+		}
+		_ = checkProcessFileData(sumResp)
+
+	}
+
+	return ms, nil
+
+}
+
+func createRuntimePolicy(policyType string) MatchSpec {
+
+	if policyType == "file" {
+		ms.Rules.FileRule = &filePaths
+
+	} else {
+		ms.Rules.ProcessRule = &processPaths
+
+	}
+
+	ms.OnEvent.Severity = 1
+	ms.Description.Tldr = "Kubernetes serviceaccount folder access should be limited "
+	ms.Precondition = "/run/secrets/kubernetes.io/serviceaccount"
+	ms.Name = "audit-serviceaccount-runtime"
+	ms.OnEvent.Tags = []string{"KUBERNETES", "SERVICE ACCOUNT", "RUNTIME POLICY"}
+	ms.OnEvent.Message = "serviceaccount access detected"
+	ms.OnEvent.Action = "Audit"
+
+	return ms
+
+}
+
+func checkProcessFileData(sumResp *opb.Response) error {
+	processPaths = SysRule{}
+	filePaths = SysRule{}
+	if len(sumResp.ProcessData) > 0 {
+
+		for _, procData := range sumResp.ProcessData {
```

Comment:
Why are we checking the process data? Checking the file access data wouldn't be sufficient?